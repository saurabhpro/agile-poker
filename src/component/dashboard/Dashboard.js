import React from 'react';
import HomePage from './HomePage';
import sessionStoreUserName from '../utils/sessionStore';
import SignIn from '../auth/SignIn';

export default function Dashboard() {
  const userName = sessionStoreUserName();

  return (
    <div>
      {userName ? <HomePage userName={userName} /> : <SignIn />}
    </div>
  );
}
