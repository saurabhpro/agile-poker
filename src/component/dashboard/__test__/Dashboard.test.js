import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from '../Dashboard';
import {
  addUserNameToSessionStore,
  sessionStoreClearUserName,
} from '../../utils/sessionStore';

let component;

describe(' dashboard without authentication', () => {
  beforeAll(() => {
    component = shallow(<Dashboard />);
    console.debug(component.debug());
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render one <SignIn>', () => {
    expect(component.find('SignIn')).toHaveLength(1);
  });
});

describe(' dashboard with authentication', () => {
  beforeAll(() => {
    addUserNameToSessionStore('Saurabh');
    component = shallow(<Dashboard />);
    //console.debug(component.debug());
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render one <HomePage>', () => {
    expect(component.find('HomePage')).toHaveLength(1);
  });

  it('should render HomePage with current user', () => {
    expect(component.find('Saurabh')).toBeTruthy();
  });

  afterAll(() => {
    sessionStoreClearUserName();
  });
});
