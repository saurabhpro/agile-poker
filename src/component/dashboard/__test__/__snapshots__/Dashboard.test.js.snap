// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` dashboard with authentication renders correctly 1`] = `
<div>
  <HomePage
    userName="Saurabh"
  />
</div>
`;

exports[` dashboard without authentication renders correctly 1`] = `
<div>
  <SignIn />
</div>
`;
